### PR TITLE
Dev-Z

### PR DESCRIPTION
**Pull request by Zeighe**
help

### CODE REVIEW

File: src/main/java/io/xeros/model/entity/npc/interactions/impl/alkharid/duelarena/SurgeonGeneralTafani.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+public class SurgeonGeneralTafani extends NpcOptionAction {
+    @Override
+    protected Integer[] getIds() {
+        return new Integer[] { SURGEON_GENERAL_TAFANI };
+    }
+
+    @Override
+    public Boolean handleActionOne(Player player, NPC npc) {
+        PestControl.refreshPlayer(player);
+        player.getDH().sendItemStatement("Restored your HP, Prayer, Run Energy, and Spec", 4049);
```
Comment (Glabay) [inline src/main/java/io/xeros/model/entity/npc/interactions/impl/alkharid/duelarena/SurgeonGeneralTafani.java]: Magic number can be replaced with Items.ITEM_NAME

File: src/main/java/io/xeros/model/entity/npc/interactions/impl/canifis/Mazchna.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+public class Mazchna extends NpcOptionAction {
+    @Override
+    protected Integer[] getIds() {
+        return new Integer[] { MAZCHNA };
+    }
+
+    @Override
+    public Boolean handleActionOne(Player player, NPC npc) {
+        if (player.combatLevel < 20) {
+            player.getDH().sendNpcChat2("Do not waste my time peasent.", "You need a Combat level of 20.", 402, "Mazchna");
```
Comment (Glabay) [inline src/main/java/io/xeros/model/entity/npc/interactions/impl/canifis/Mazchna.java]: NPC ID 402 can be replaced with the MAZCHNA

File: src/main/java/io/xeros/model/entity/npc/interactions/impl/canifis/Mazchna.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Override
+    public Boolean handleActionTwo(Player player, NPC npc) {
+        if (player.fullVoidMelee()) {
+            player.getDiaryManager().getWesternDiary().progress(WesternDiaryEntry.FULL_VOID);
+        }
+        if (player.getSlayer().getTask().isPresent()) {
+            player.getDH().sendDialogues(3305, MAZCHNA); //Glabay and Z to review dialogue 3305 and 180 as a  QOL would allow all masters to reset your task with an easy one
+        } else {
+            player.getDH().sendDialogues(180, MAZCHNA);
```
Comment (Glabay) [inline src/main/java/io/xeros/model/entity/npc/interactions/impl/canifis/Mazchna.java]: the Dialogue ID can be added to the Dialogues.java class

File: src/main/java/io/xeros/model/entity/npc/interactions/impl/global/MasterFarmer.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            if (Boundary.isIn(player, Boundary.FALADOR_BOUNDARY)) {
+                player.getThieving().steal(Thieving.Pickpocket.FARMER, npc);
+            }
+            if (Boundary.isIn(player, Boundary.DRAYNOR_BOUNDARY)) {
+                player.getThieving().steal(Thieving.Pickpocket.FARMER, npc);
+                player.getDiaryManager().getLumbridgeDraynorDiary().progress(LumbridgeDraynorDiaryEntry.PICKPOCKET_FARMER_DRAY);
+            }
+            player.sendMessage("@red@Restricted accounts can only steal from this npc.");
+        } else {
+            player.getShops().openShop(16);
```
Comment (Glabay) [inline src/main/java/io/xeros/model/entity/npc/interactions/impl/global/MasterFarmer.java]: Perhaps we can make a `Shops.java` to hold the shop IDs to not have Magic numbers... Hmm

File: src/main/java/io/xeros/model/entity/npc/interactions/impl/tutorialisland/CombatInstructor.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Override
+    public Boolean handleActionOne(Player player, NPC npc) {
+        player.getDH().sendDialogues(COMBAT_INSTRUCTOR_DIALOGUE, COMBAT_INSTRUCTOR);
+        return true;
+    }
+
+    @Override
+    public Boolean handleActionTwo(Player player, NPC npc) {
+        player.getPA().showInterface(37700);
```
Comment (Glabay) [inline src/main/java/io/xeros/model/entity/npc/interactions/impl/tutorialisland/CombatInstructor.java]: Hmm maybe we need an `Interfaces.java` for these Magic Numbers

**Review by Glabay — CHANGES_REQUESTED** [review]

File: src/main/java/io/xeros/model/entity/npc/interactions/impl/taverly/dungeon/KeyMaster.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+            if (player.getSlayer().getTask().isPresent()) {
+                player.getDH().sendStatement("Please finish your current task first.");
+                return false;
+            }
+            if (!player.getItems().playerHasItem(COINS, 3_000_000)) {
+                player.getDH().sendStatement("Come back when you've got the 3m ready.");
+                return false;
+            }
+            player.getItems().deleteItem2(COINS, 3_000_000);
+            player.getSlayer().createNewTask(5870, true);
```
Comment (Glabay) [inline src/main/java/io/xeros/model/entity/npc/interactions/impl/taverly/dungeon/KeyMaster.java]: This is a Magic Number

**Review by Glabay — CHANGES_REQUESTED** [review]

File: src/main/java/io/xeros/content/bosses/grotesqueguardians/GrotesqueInstance.java [hunk 7]
```diff
@@ -37,8 +37,8 @@ public GrotesqueInstance() {
     public void enter(Player plr) {
         try {
             add(plr);
-            GrotesqueGuardianNpc dawn = new GrotesqueGuardianNpc(Npcs.DAWN_2, new Position(1700, 4576, getHeight()), this);
-            GrotesqueGuardianNpc dusk = new GrotesqueGuardianNpc(Npcs.DUSK_2, new Position(1692, 4576, getHeight()), this);
```
Comment (Glabay) [inline src/main/java/io/xeros/content/bosses/grotesqueguardians/GrotesqueInstance.java:41]: These are preparing the 2 Bosses, Dusk and Dawn, and you have the same ID for both here

File: src/main/java/io/xeros/content/bosses/grotesqueguardians/GrotesqueGuardianNpc.java [hunk 8]
```diff
@@ -194,9 +194,9 @@ public void process() {
                     counterpart.setInvisible(false);
                     counterpart.teleport(1701, 4573,getHeight());
                     counterpart.startAnimation(7774);
-                    counterpart.requestTransform(Npcs.DAWN_3);
+                    counterpart.requestTransform(Npcs.DAWN_7853);
                     //stop dusk
-                    requestTransform(Npcs.DUSK_4);
```
Comment (Glabay) [inline src/main/java/io/xeros/content/bosses/grotesqueguardians/GrotesqueGuardianNpc.java:199]: This ID Changed from 7854 to 7855, was this intentional?

File: src/main/java/io/xeros/content/bosses/grotesqueguardians/GrotesqueGuardianNpc.java [hunk 9]
```diff
@@ -130,7 +130,7 @@ public boolean canBeAttacked(Entity entity) {
                 p.sendMessage("Dusk is defending himself with his wing! He absorbs the attack.");
             }
         }
-        if (getNpcId() == Npcs.DAWN_3 || getNpcId() == Npcs.DUSK_4)
```
Comment (Glabay) [inline src/main/java/io/xeros/content/bosses/grotesqueguardians/GrotesqueGuardianNpc.java:133]: DUSK_4 changed from 7854 to 7855

File: src/main/java/io/xeros/model/entity/npc/interactions/impl/global/makeovermage.java [hunk 10]
```diff
@@ -16,7 +17,7 @@
 public class makeovermage extends NpcOptionAction {
```
Comment (Glabay) [inline src/main/java/io/xeros/model/entity/npc/interactions/impl/global/makeovermage.java:16]: The Name of this class is wrong, should be CameCase  `MakeoverMage`

File: src/main/java/io/xeros/model/entity/npc/interactions/impl/global/makeovermage.java [hunk 11]
```diff
@@ -26,7 +27,7 @@ public Boolean handleActionOne(Player player, NPC npc) {
             player.canChangeAppearance = false;
         }
         else {
-            player.getPA().showInterface(3559);
```
Comment (Glabay) [inline src/main/java/io/xeros/model/entity/npc/interactions/impl/global/makeovermage.java:29]: Looks like you replaced an Interface.MAKEOVER_MAGE_INTERFACE with a Magic Number? Why?

File: src/main/java/io/xeros/model/entity/npc/interactions/impl/lumbridge/LumbridgeGuide.java [hunk 12]
```diff
@@ -25,7 +26,7 @@ public Boolean handleActionOne(Player player, NPC npc) {
             player.sendMessage("You cannot do this right now.");
             return false;
         }
-        player.getDH().sendDialogues(710, 306);
```
Comment (Glabay) [inline src/main/java/io/xeros/model/entity/npc/interactions/impl/lumbridge/LumbridgeGuide.java:28]: Looks like you removed the Constants and added Magic Numbers

used to be `LUMBRIDGE_GUIDE_DIALOGUE, LUMBRIDGE_GUIDE`

**Review by Glabay — CHANGES_REQUESTED** [review]

File: src/main/java/io/xeros/content/bosses/grotesqueguardians/GrotesqueGuardianNpc.java (hunk 8 above)
Comment (Zeighe) [reply src/main/java/io/xeros/content/bosses/grotesqueguardians/GrotesqueGuardianNpc.java:199]: Yeah bb

**Review by Glabay — APPROVED** [review]

**Pull request merged**